shapely/geometry/geo.py

```diff
@@ -90,8 +113,31 @@ def shape(context):
 
 
 def asShape(context):
-    """Adapts the context to a geometry interface. The coordinates remain
-    stored in the context.
+    """
+    Adapts the context to a geometry interface. The coordinates remain
+    stored in the context, and changes to them will be reflected in the
+    returned geometry object.
+
+    Parameters
+    ----------
+    context :
+        a GeoJSON-like dict, which provides a "type" member describing the type
+        of the geometry and "coordinates" member providing a list of coordinates,
+        or an object which implements __geo_interface__.
+
```

Comment:
@tomplex shape is almost always better than asShape (see https://github.com/Toblerity/Shapely/issues/124) and we're going to start warning about asShape in the next version. What would you think about adding the following?

```suggestion

Notes
-----
The Adapter classes returned by this function trade performance for reduced storage of coordinate values. In general, the shape() function should be used instead.

```

Reply:
No problem! I didn't know that myself, so I'm glad you mentioned it.